flink-libraries/flink-table/src/main/scala/org/apache/flink/table/catalog/ExternalCatalogTable.scala

```diff
@@ -37,25 +36,10 @@ import org.apache.flink.table.plan.stats.TableStats
   * @param lastAccessTime       Timestamp of last access of the table
   */
 case class ExternalCatalogTable(
-    identifier: TableIdentifier,
```

Comment:
If we don't have table identifier or table name, the `TableSourceConverter` will not work when trying to convert `ExternalCatalogTable` to a real `TableSource`.

Even if we keep the table name, i'm not sure it will work for all situations. For example, if we have an outside catalog which have the notion of database, like MySQL. Table in that catalog will named as "db1.table1" or "db2.table2". In the new design, normally we should add database as the sub catalog of the root catalog. So we can create "table1" from sub-catalog named "db1". The problem is the table name is assigned as "table1", we loose the information that this table actually comes from "db1". This may cause some problems when the TableSource trying to establish connection or trying to get table information from outside catalog.